hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -128,14 +128,35 @@ object HoodieSparkSqlWriter {
           .setPayloadClassName(hoodieConfig.getString(PAYLOAD_CLASS_OPT_KEY))
           .setPreCombineField(hoodieConfig.getStringOrDefault(PRECOMBINE_FIELD_OPT_KEY, null))
           .setPartitionColumns(partitionColumns)
+          .setPopulateMetaColumns(parameters.getOrElse(HoodieTableConfig.HOODIE_POPULATE_META_COLUMNS.key(), HoodieTableConfig.HOODIE_POPULATE_META_COLUMNS.defaultValue()).toBoolean)
           .initTable(sparkContext.hadoopConfiguration, path.get)
         tableConfig = tableMetaClient.getTableConfig
+      } else {
+        // validate table properties
+        val tableMetaClient = HoodieTableMetaClient.builder().setBasePath(path.get).setConf(sparkContext.hadoopConfiguration).build()
```

Comment:
is there other places where we do such validation. I think this should be done at the WriteClient level or so? not this high up the stack?

Reply:
In any case, please pull all this validation into its own method.

Reply:
I am thinking to add the validation within HoodieTableMetaClient. Because, a) We don't instantiate WriteClient at all in row writer path as of now. b) table properties are available when tableMetaclient is instantiated. Even if we were to place it within WriteClient, we have to instantiate Metaclient to read the table props. So, we might as well place the validation method in HoodieTableMetaClient.

Reply:
Added a private method here in HoodiesparkSqlWriter wrt params and add a method in HoodieTableMetaclient to validate table properties.

Reply:
I understand that we need one place to do the validation. 
As of now, any callers directly to WriteClient has to make another call to do validation. 

I could think of another option. Basically fix all startCommit() methods in WriteClient to take in operationType and do the validation. But there are quite a few callers which I need to fix. If we go ahead with this approach, validation happens within our custom data source where we instantiate the writeClient and start the commit (in row writer path). 
Once we have consensus I can make the changes. I feel this is the right approach.